File: android/app/src/main/java/com/talentlens/app/ui/screens/BenchmarksScreen.kt
package com.talentlens.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.talentlens.app.engine.PercentileEngine
import com.talentlens.app.model.ExerciseType
import com.talentlens.app.model.Gender
import com.talentlens.app.ui.theme.*

@Composable
fun BenchmarksScreen() {
    var selectedExercise by remember { mutableStateOf(ExerciseType.PUSHUPS) }
    var selectedGender by remember { mutableStateOf(Gender.MALE) }
    var age by remember { mutableStateOf(17f) }
    var score by remember { mutableStateOf(38f) }

    val calc = remember(selectedExercise, selectedGender, age, score) {
        PercentileEngine.calculate(
            score = score.toInt(),
            age = age.toInt(),
            gender = selectedGender,
            exerciseType = selectedExercise
        )
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(BackgroundDark)
            .padding(16.dp)
            .verticalScroll(rememberScrollState()),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Header
        Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
            Text("NATIONAL STANDARDS", color = BrandOrange, fontSize = 11.sp, fontWeight = FontWeight.Bold, fontFamily = FontFamily.Monospace)
            Text("Percentile Calculator", color = TextPrimary, fontSize = 24.sp, fontWeight = FontWeight.Black)
            Text("Explore empirical distributions calibrated against Khelo India / SAI youth norms.", color = TextSecondary, fontSize = 12.sp)
        }

        // Exercise Tabs
        LazyRow(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            items(ExerciseType.values()) { ex ->
                val isSelected = ex == selectedExercise
                Surface(
                    color = if (isSelected) CardBackground else CardBackground.copy(alpha = 0.5f),
                    shape = RoundedCornerShape(14.dp),
                    border = androidx.compose.foundation.BorderStroke(
                        if (isSelected) 1.5.dp else 1.dp,
                        if (isSelected) BrandOrange else CardBorder
                    ),
                    modifier = Modifier.clickable { selectedExercise = ex }
                ) {
                    Text(
                        text = ex.shortName,
                        color = if (isSelected) TextPrimary else TextSecondary,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Bold,
                        modifier = Modifier.padding(horizontal = 14.dp, vertical = 8.dp)
                    )
                }
            }
        }

        // Gender Selector
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            listOf(Gender.MALE, Gender.FEMALE).forEach { g ->
                val isSelected = g == selectedGender
                Button(
                    onClick = { selectedGender = g },
                    colors = ButtonDefaults.buttonColors(containerColor = if (isSelected) BrandOrange else CardBackground),
                    shape = RoundedCornerShape(12.dp),
                    modifier = Modifier.weight(1f).border(1.dp, if (isSelected) BrandOrange else CardBorder, RoundedCornerShape(12.dp))
                ) {
                    Text(g.displayName, color = if (isSelected) TextPrimary else TextSecondary, fontWeight = FontWeight.Bold)
                }
            }
        }

        // Sliders Card
        Card(
            shape = RoundedCornerShape(20.dp),
            colors = CardDefaults.cardColors(containerColor = CardBackground),
            modifier = Modifier.fillMaxWidth().border(1.dp, CardBorder, RoundedCornerShape(20.dp))
        ) {
            Column(modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(16.dp)) {
                // Age Slider
                Column {
                    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                        Text("ATHLETE AGE", color = TextSecondary, fontSize = 11.sp, fontWeight = FontWeight.Bold)
                        Text("${age.toInt()} Years Old", color = TextPrimary, fontSize = 13.sp, fontWeight = FontWeight.Bold)
                    }
                    Slider(
                        value = age,
                        onValueChange = { age = it },
                        valueRange = 10f..35f,
                        colors = SliderDefaults.colors(thumbColor = BrandOrange, activeTrackColor = BrandOrange)
                    )
                }

                // Score Slider
                Column {
                    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                        Text("TEST SCORE", color = TextSecondary, fontSize = 11.sp, fontWeight = FontWeight.Bold)
                        Text("${score.toInt()} ${selectedExercise.metricUnit}", color = BrandOrange, fontSize = 13.sp, fontWeight = FontWeight.Bold)
                    }
                    Slider(
                        value = score,
                        onValueChange = { score = it },
                        valueRange = 1f..(if (selectedExercise == ExerciseType.PLANK) 300f else 100f),
                        colors = SliderDefaults.colors(thumbColor = BrandOrange, activeTrackColor = BrandOrange)
                    )
                }
            }
        }

        // Dynamic Calculation Card
        Surface(
            color = Color(calc.talentTier.badgeColorHex).copy(alpha = 0.15f),
            shape = RoundedCornerShape(20.dp),
            border = androidx.compose.foundation.BorderStroke(1.dp, Color(calc.talentTier.badgeColorHex).copy(alpha = 0.5f)),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.padding(20.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text("CALCULATED PLACEMENT", color = TextSecondary, fontSize = 10.sp, fontWeight = FontWeight.Bold, fontFamily = FontFamily.Monospace)
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.Bottom
                ) {
                    Text("${calc.percentileRounded}%", color = TextPrimary, fontSize = 42.sp, fontWeight = FontWeight.Black)
                    Text(calc.talentTier.displayName, color = Color(calc.talentTier.badgeColorHex), fontSize = 12.sp, fontWeight = FontWeight.Bold)
                }
                Text(calc.comparisonSummary, color = TextSecondary, fontSize = 12.sp)
            }
        }
    }
}

File: ios/TalentLens/UI/Screens/BenchmarksScreen.swift
//
//  BenchmarksScreen.swift
//  TalentLens
//
//  Created for TalentLens Olympic-Grade Athletic Testing Suite.
//

import SwiftUI

public struct BenchmarksScreen: View {
    @State private var selectedTab: BenchmarkTab = .proChampions
    @State private var selectedExercise: ExerciseType = .pushups
    @State private var selectedGender: Gender = .male
    @State private var age: Double = 17.0
    @State private var score: Double = 38.0
    
    public enum BenchmarkTab: String, CaseIterable, Identifiable {
        case proChampions = "⭐ Pro Champions Dataset"
        case nationalPercentiles = "📊 SAI National Percentiles"
        
        public var id: String { rawValue }
    }
    
    public init() {}
    
    private var calculation: PercentileCalculation {
        PercentileEngine.calculate(
            score: Int(score),
            age: Int(age),
            gender: selectedGender,
            exerciseType: selectedExercise
        )
    }
    
    public var body: some View {
        ZStack {
            TLTheme.backgroundDark.edgesIgnoringSafeArea(.all)
            
            ScrollView {
                VStack(alignment: .leading, spacing: 16) {
                    // Header
                    VStack(alignment: .leading, spacing: 4) {
                        Text("STANDARDS & PRO BENCHMARKS")
                            .font(.system(size: 11, weight: .bold, design: .monospaced))
                            .foregroundColor(TLTheme.brandOrange)
                        Text("Elite Athletic Datasets")
                            .font(.system(size: 24, weight: .black))
                            .foregroundColor(TLTheme.textPrimary)
                        Text("Benchmark training metrics against Olympic champions and SAI national youth norms.")
                            .font(.system(size: 12))
                            .foregroundColor(TLTheme.textSecondary)
                    }
                    .padding(.horizontal)
                    .padding(.top, 12)
                    
                    // Tab Selector (Pro Champions vs SAI Percentiles)
                    HStack(spacing: 8) {
                        ForEach(BenchmarkTab.allCases) { tab in
                            let isSelected = (tab == selectedTab)
                            Button(action: { selectedTab = tab }) {
                                Text(tab.rawValue)
                                    .font(.system(size: 12, weight: .bold))
                                    .frame(maxWidth: .infinity)
                                    .padding(.vertical, 10)
                                    .background(isSelected ? TLTheme.brandOrange : TLTheme.cardBackground)
                                    .foregroundColor(isSelected ? .white : TLTheme.textSecondary)
                                    .cornerRadius(14)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 14)
                                            .stroke(isSelected ? TLTheme.brandOrange : TLTheme.cardBorder, lineWidth: 1)
                                    )
                            }
                        }
                    }
                    .padding(.horizontal)
                    
                    if selectedTab == .proChampions {
                        // Pro Champions Dataset View
                        VStack(alignment: .leading, spacing: 14) {
                            Text("WORLD-CLASS & OLYMPIC PRO DATASET (11 CHAMPIONS)")
                                .font(.system(size: 10, weight: .bold, design: .monospaced))
                                .foregroundColor(TLTheme.cyberCyan)
                                .padding(.horizontal)
                            
                            ForEach(ProAthleteDataset.athletes) { pro in
                                VStack(alignment: .leading, spacing: 10) {
                                    HStack(spacing: 12) {
                                        ZStack {
                                            Circle()
                                                .fill(TLTheme.brandOrange.opacity(0.15))
                                                .frame(width: 48, height: 48)
                                            Text(pro.iconEmoji)
                                                .font(.title2)
                                        }
                                        
                                        VStack(alignment: .leading, spacing: 2) {
                                            HStack {
                                                Text(pro.name)
                                                    .font(.system(size: 16, weight: .black))
                                                    .foregroundColor(TLTheme.textPrimary)
                                                Spacer()
                                                Text(pro.sport.iconEmoji)
                                                    .font(.subheadline)
                                                Text(pro.sport.displayName.components(separatedBy: " (").first ?? "")
                                                    .font(.system(size: 10, weight: .bold))
                                                    .foregroundColor(TLTheme.brandOrange)
                                            }
                                            
                                            Text(pro.title)
                                                .font(.system(size: 12, weight: .semibold))
                                                .foregroundColor(TLTheme.cyberCyan)
                                            
                                            Text(pro.achievement)
                                                .font(.system(size: 10))
                                                .foregroundColor(TLTheme.textSecondary)
                                        }
                                    }
                                    
                                    Divider().background(TLTheme.cardBorder)
                                    
                                    // 4-Pillar Pro Benchmark Matrix
                                    HStack(spacing: 6) {
                                        ProStatBadge(label: "PUSH-UPS", val: "\(pro.pushupsScore) reps")
                                        ProStatBadge(label: "SQUATS", val: "\(pro.squatsScore) reps")
                                        ProStatBadge(label: "PLANK", val: "\(pro.plankSeconds)s")
                                        ProStatBadge(label: "JUMP", val: "\(Int(pro.verticalJumpCm)) cm")
                                    }
                                    
                                    // Advice & Archetype
                                    VStack(alignment: .leading, spacing: 4) {
                                        Text("Archetype: \(pro.physicalArchetype)")
                                            .font(.system(size: 10, weight: .semibold))
                                            .foregroundColor(TLTheme.textPrimary)
                                        
                                        Text("\"\(pro.proAdviceQuote)\"")
                                            .font(.system(size: 10))
                                            .italic()
                                            .foregroundColor(TLTheme.textSecondary)
                                            .lineLimit(2)
                                    }
                                    .padding(8)
                                    .background(TLTheme.backgroundDark.opacity(0.7))
                                    .cornerRadius(8)
                                }
                                .padding(16)
                                .background(TLTheme.cardBackground)
                                .cornerRadius(20)
                                .overlay(RoundedRectangle(cornerRadius: 20).stroke(TLTheme.cardBorder, lineWidth: 1))
                                .padding(.horizontal)
                            }
                        }
                    } else {
                        // National Percentiles View
                        // Exercise Tabs
                        ScrollView(.horizontal, showsIndicators: false) {
                            HStack(spacing: 8) {
                                ForEach(ExerciseType.allCases) { ex in
                                    let isSelected = (ex == selectedExercise)
                                    Button(action: {
                                        selectedExercise = ex
                                        if ex == .plank && score < 60 {
                                            score = 120
                                        } else if ex != .plank && score > 100 {
                                            score = 40
                                        }
                                    }) {
                                        VStack(alignment: .leading, spacing: 2) {
                                            Text(ex.category.uppercased())
                                                .font(.system(size: 9, weight: .bold))
                                                .foregroundColor(isSelected ? TLTheme.brandOrange : TLTheme.textSecondary)
                                            Text(ex.shortName)
                                                .font(.system(size: 13, weight: .bold))
                                                .foregroundColor(TLTheme.textPrimary)
                                        }
                                        .padding(.horizontal, 14)
                                        .padding(.vertical, 8)
                                        .background(isSelected ? TLTheme.cardBackground : TLTheme.cardBackground.opacity(0.4))
                                        .cornerRadius(12)
                                        .overlay(
                                            RoundedRectangle(cornerRadius: 12)
                                                .stroke(isSelected ? TLTheme.brandOrange : TLTheme.cardBorder, lineWidth: isSelected ? 1.5 : 1)
                                        )
                                    }
                                }
                            }
                            .padding(.horizontal)
                        }
                        
                        // Interactive Sliders Card
                        VStack(spacing: 16) {
                            // Gender Segmented Picker
                            Picker("Gender", selection: $selectedGender) {
                                ForEach(Gender.allCases) { g in
                                    Text(g.displayName).tag(g)
                                }
                            }
                            .pickerStyle(SegmentedPickerStyle())
                            
                            // Age Slider
                            VStack(alignment: .leading, spacing: 6) {
                                HStack {
                                    Text("Athlete Age")
                                        .font(.system(size: 12, weight: .bold))
                                        .foregroundColor(TLTheme.textSecondary)
                                    Spacer()
                                    Text("\(Int(age)) years")
                                        .font(.system(size: 14, weight: .bold, design: .monospaced))
                                        .foregroundColor(TLTheme.textPrimary)
                                }
                                Slider(value: $age, in: 10...35, step: 1)
                                    .accentColor(TLTheme.brandOrange)
                            }
                            
                            // Score Slider
                            VStack(alignment: .leading, spacing: 6) {
                                HStack {
                                    Text(selectedExercise.title)
                                        .font(.system(size: 12, weight: .bold))
                                        .foregroundColor(TLTheme.textSecondary)
                                    Spacer()
                                    Text("\(Int(score)) \(selectedExercise.metricUnit)")
                                        .font(.system(size: 14, weight: .bold, design: .monospaced))
                                        .foregroundColor(TLTheme.brandOrange)
                                }
                                Slider(
                                    value: $score,
                                    in: 1...(selectedExercise == .plank ? 300.0 : 100.0),
                                    step: 1
                                )
                                .accentColor(TLTheme.brandOrange)
                            }
                        }
                        .padding(18)
                        .background(TLTheme.cardBackground)
                        .cornerRadius(20)
                        .overlay(RoundedRectangle(cornerRadius: 20).stroke(TLTheme.cardBorder, lineWidth: 1))
                        .padding(.horizontal)
                        
                        // Dynamic Calculation Card
                        let calc = calculation
                        VStack(alignment: .leading, spacing: 10) {
                            Text("CALCULATED PLACEMENT")
                                .font(.system(size: 10, weight: .bold, design: .monospaced))
                                .foregroundColor(TLTheme.textSecondary)
                            
                            HStack(alignment: .lastTextBaseline) {
                                Text("\(calc.percentileRounded)%")
                                    .font(.system(size: 42, weight: .black))
                                    .foregroundColor(TLTheme.textPrimary)
                                Spacer()
                                Text(calc.talentTier.displayName)
                                    .font(.system(size: 12, weight: .bold))
                                    .foregroundColor(Color(hex: calc.talentTier.badgeColorHex))
                            }
                            
                            Text(calc.comparisonSummary)
                                .font(.system(size: 12))
                                .foregroundColor(TLTheme.textSecondary)
                        }
                        .padding(20)
                        .background(Color(hex: calc.talentTier.badgeColorHex).opacity(0.15))
                        .cornerRadius(20)
                        .overlay(
                            RoundedRectangle(cornerRadius: 20)
                                .stroke(Color(hex: calc.talentTier.badgeColorHex).opacity(0.5), lineWidth: 1)
                        )
                        .padding(.horizontal)
                    }
                    
                    Spacer().frame(height: 24)
                }
            }
        }
    }
}

private struct ProStatBadge: View {
    let label: String
    let val: String
    
    var body: some View {
        VStack(spacing: 2) {
            Text(label)
                .font(.system(size: 8, weight: .bold, design: .monospaced))
                .foregroundColor(TLTheme.textSecondary)
            Text(val)
                .font(.system(size: 10, weight: .black, design: .monospaced))
                .foregroundColor(TLTheme.brandOrange)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 6)
        .background(TLTheme.backgroundDark.opacity(0.8))
        .cornerRadius(8)
    }
}
